core/src/main/scala/org/broadinstitute/dsde/rawls/snapshot/SnapshotService.scala

```diff
@@ -146,12 +146,15 @@ class SnapshotService(protected val userInfo: UserInfo, val dataSource: SlickDat
     SnapshotListResponse(refs)
   }
 
-  def updateSnapshot(workspaceName: WorkspaceName, snapshotId: String, updateInfo: UpdateDataReferenceRequestBody): Future[Unit] = {
+  def updateSnapshot(workspaceName: WorkspaceName, snapshotId: String, updateInfo: UpdateDataRepoSnapshotReferenceRequestBody): Future[Unit] = {
     val snapshotUuid = validateSnapshotId(snapshotId)
     getWorkspaceContextAndPermissions(workspaceName, SamWorkspaceActions.write, Some(WorkspaceAttributeSpecs(all = false))).map { workspaceContext =>
       // check that snapshot exists before updating it. If the snapshot does not exist, the GET attempt will throw a 404
       workspaceManagerDAO.getDataRepoSnapshotReference(workspaceContext.workspaceIdAsUUID, snapshotUuid, userInfo.accessToken)
-      // build the update request body
+      // build the update request body, ignoring any changes to instanceName and snapshot, and requiring either name or description
+      if (Option(updateInfo.getName).isEmpty && Option(updateInfo.getDescription).isEmpty) {
+        throw new RawlsExceptionWithErrorReport(ErrorReport(StatusCodes.BadRequest, "Either name or description is required."))
+      }
```

Comment:
`UpdateDataReferenceRequestBody`, which no longer exists in the WSM client, had only "name" and "description". `UpdateDataRepoSnapshotReferenceRequestBody`, which we're using now, has "name", "description", "instanceName", and "snapshot". I added this clause so our behavior is the same as previous. We don't have a use case currently for updating the instanceName or snapshot.